File: implementation/csharp/tests/GameEventScript.Tests/HostTestStartup.cs
// Copyright 2026 Stephan Schlöpke
// SPDX-License-Identifier: Apache-2.0

using GameEventScript.Api;

namespace GameEventScript.Tests;

internal static class HostTestStartup
{
    internal static GameEventScriptHost StartForTest(this GameEventScriptHost host)
    {
        Assert.AreEqual(GameEventScriptStartState.Ready, host.Start().State);
        return host;
    }
}

File: implementation/swift/GameEventScriptConformance/Tests/GameEventScriptConformanceTests/HostTestStartup.swift
// Copyright 2026 Stephan Schlöpke
// SPDX-License-Identifier: Apache-2.0

import GameEventScriptRuntime
import XCTest

extension GameEventScriptHost {
    func startForTest() throws -> Self {
        XCTAssertEqual(try start().state, .ready)
        return self
    }
}
